_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

```diff
@@ -924,38 +946,25 @@ mfxStatus VideoDECODEAV1::SubmitFrame(mfxBitstream* bs, mfxFrameSurface1* surfac
                     sts = MFX_ERR_GPU_HANG;
                     break;
 #endif
-            }
 
-            if (umcRes == UMC::UMC_WRN_REPOSITION_INPROGRESS)
-            {
-                if (!m_first_run)
-                {
-                    sts = MFX_WRN_VIDEO_PARAM_CHANGED;
-                }
-                else
-                {
-                    umcRes = UMC::UMC_OK;
-                    m_first_run = false;
-                }
-            }
-
-            if (umcRes == UMC::UMC_OK && m_allocator->FindFreeSurface() == -1)
-            {
-                sts = MFX_ERR_MORE_SURFACE;
-                umcFrameRes = UMC::UMC_ERR_NOT_ENOUGH_BUFFER;
-            }
+                case UMC::UMC_ERR_NOT_ENOUGH_BUFFER:
+                case UMC::UMC_WRN_INFO_NOT_READY:
+                case UMC::UMC_ERR_NEED_FORCE_OUTPUT:
+                    force = (umcRes == UMC::UMC_ERR_NEED_FORCE_OUTPUT);
+                    sts = umcRes == UMC::UMC_ERR_NOT_ENOUGH_BUFFER ? (mfxStatus)MFX_ERR_MORE_DATA_SUBMIT_TASK: MFX_WRN_DEVICE_BUSY;
+                    break;
 
-            if (umcRes == UMC::UMC_ERR_NOT_ENOUGH_BUFFER || umcRes == UMC::UMC_WRN_INFO_NOT_READY || umcRes == UMC::UMC_ERR_NEED_FORCE_OUTPUT)
-            {
-                force = (umcRes == UMC::UMC_ERR_NEED_FORCE_OUTPUT);
-                sts = umcRes == UMC::UMC_ERR_NOT_ENOUGH_BUFFER ? (mfxStatus)MFX_ERR_MORE_DATA_SUBMIT_TASK: MFX_WRN_DEVICE_BUSY;
-            }
+                case UMC::UMC_ERR_NOT_ENOUGH_DATA:
+                case UMC::UMC_ERR_SYNC:
+                    if (!bs || bs->DataFlag == MFX_BITSTREAM_EOS)
+                        force = true;
+                    sts = MFX_ERR_MORE_DATA;
+                    break;
 
-            if (umcRes == UMC::UMC_ERR_NOT_ENOUGH_DATA || umcRes == UMC::UMC_ERR_SYNC)
-            {
-                if (!bs || bs->DataFlag == MFX_BITSTREAM_EOS)
-                    force = true;
-                sts = MFX_ERR_MORE_DATA;
+                default:
+                    if (sts < 0)
+                        sts = MFX_ERR_UNKNOWN;
+                    break;
             }
 
 
```

Comment:
github does not show lines below, but they are:
```
if (umcRes != UMC::UMC_NTF_NEW_RESOLUTION)
{...}
```

This condition does not make sense to me since it is always true, isn't it? (we have return checking this status in switch)

Reply:
Fair enough. Removed unnecessary condition.